src/frameworks/xtk.js

```diff
@@ -15,13 +15,8 @@ export class Xtk extends Framework {
     let image = ctx.getImageData(0, 0, canvas.width, canvas.height);
     let rgba_image = Util.rgba_to_grayscale(image.data);
 
-    if(from_canvas) {
-      return { data: image.data, width: image.width, height: image.height };
-    }
-    else {
-      return { data: rgba_image, width: image.width, height: image.height };
+    return { data: rgba_image, width: image.width, height: image.height };
```

Comment:
Hello @shrutivarade, why did you get rid of the different return features? If I remember correctly, this allowed it to work correctly, just out of curiosity :)